### PR TITLE
Handle case when `qtys` param is null when preparing shipment

### PR DESCRIPTION
**Pull request by CharlieDelta6**
PHP8 fix - Uncaught TypeError if $qtys is NULL

### Fixed Issues (if relevant)

1. Fixes OpenMage/magento-lts#<issue_number>

### Manual testing scenarios (*)

1. Install stamps.com integration
2. Call update on integration script: /Stamps.php?username=[Stamps Username]&password=[Password]&action=updateorder&order=[order number]&comments=&command=complete&tracking=[tracking number]&carrier=[carrier]
3. Error triggered when script attempts to obtain order quantities & create shipment

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

**Comment by fballiano** [comment]
if I rebase on branch 1.9.4.x I get 5000 modified files, but I think this PR should be applied to 1.9.4.x :-)

**Comment by elidrissidev** [comment]
Would also be good to include reproduction steps in OP.

File: app/code/core/Mage/Sales/Model/Service/Order.php [hunk 1]
```diff
@@ -189,7 +189,7 @@ public function prepareShipment($qtys = [])
             } else {
                 if (isset($qtys[$orderItem->getId()])) {
                     $qty = min($qtys[$orderItem->getId()], $orderItem->getQtyToShip());
-                } elseif (!count($qtys)) {
+                } elseif (is_null($qtys) || !count($qtys)) {
```
Comment (elidrissidev) [inline app/code/core/Mage/Sales/Model/Service/Order.php:192]: Can be replaced with `empty` which will check both conditions:
```suggestion
                } elseif (empty($qtys)) {
```

```php
$qtys = null;
var_dump(is_null($qtys) || !count($qtys)); // bool(true)
$qtys = [];
var_dump(is_null($qtys) || !count($qtys)); // bool(true)

$qtys = null;
var_dump(empty($qtys)); // bool(true)
$qtys = [];
var_dump(empty($qtys)); // bool(true)
```

**Comment by CharlieDelta6** [comment]
Whoops, clarified the trigger in the original request.

If using the stamps.com Magento 1 integration, the stamps.php file for Magento 1 can/will trigger this error when creating shipments.

**Comment by elidrissidev** [comment]
> Whoops, clarified the trigger in the original request.
> 
> If using the stamps.com Magento 1 integration, the stamps.php file for Magento 1 can/will trigger this error when creating shipments.

Seems like the issue is with the extension then, as `$qtys` must always be an array. But it's an easy fix anyways so I don't mind.

**Review by elidrissidev — APPROVED** [review]

**Review by fballiano — APPROVED** [review]

**Comment by fballiano** [comment]
merged and v20ed

**Comment by CharlieDelta6** [comment]
Thanks everyone, apologies if the pull request was vague. I'm an agricultural-grade coder, field fixing problems with whatever crummy toolset I have .

**Pull request merged**